ptex-sys/tests/integration_test.rs

```diff
@@ -93,3 +94,53 @@ fn funky_values2() {
         Ok("PtexWriter error: Invalid data type")
     );
 }
+
+fn make_test_writer() -> *mut PtexWriter {
+    let filename = "out.ptx";
+    let alpha_channel = -1;
+    let num_channels = 3;
+    let num_faces = 9;
+    let genmipmaps = false;
+    let meshtype = ptex_sys::MeshType::Quad;
+    let datatype = ptex_sys::DataType::UInt8;
+
+    let_cxx_string!(error_str = "");
+
+    let writer = unsafe {
+        ptex_sys::ptexwriter_open(
+            filename,
+            meshtype,
+            datatype,
+            num_channels,
+            alpha_channel,
+            num_faces,
+            genmipmaps,
+            error_str.as_mut().get_unchecked_mut(),
+        )
+    };
+    assert!(!writer.is_null());
+    writer
+}
+
+#[test]
+fn funky_values_edge_id() {
+    let writer = make_test_writer();
+    let mut face_info = FaceInfo::default();
+    // Left currently being the last variant in `EdgeId`.
+    let mut e1 = EdgeId::Left;
+    let mut e2 = EdgeId::Left;
+    let mut e3 = EdgeId::Left;
+    let mut e4 = EdgeId::Left;
+    e1.repr += 1;
+    e2.repr += 2;
+    e3.repr += 3;
+    e4.repr += 4;
+    let stride = 0;
+    face_info.set_adjacent_edges(e1, e2, e3, e4);
+    let data = [0_u8; 9];
+    let wrote_face = unsafe {
+        ptexwriter_write_face(writer, 0, &face_info, data.as_ptr(), stride)
+    };
+    // Currently this fails, as it seems it is possible to write funky values in place of EdgeIds.
+    assert_eq!(wrote_face, false);
```

Comment:
I think we can change this to `assert!(wrote_face);` since this helps to cement down the expected C++ Ptex behavior.

Ptex takes the approach of truncating out of band inputs for the adjacent edge metadata.

```cpp
struct FaceInfo {
    Res res;		///< Resolution of face.
    uint8_t adjedges;       ///< Adjacent edges, 2 bits per edge.
...
    /// Set the adjedges data.
    void setadjedges(int e0, int e1, int e2, int e3)
    { adjedges = (uint8_t)((e0&3) | ((e1&3)<<2) | ((e2&3)<<4) | ((e3&3)<<6)); }
```

Ptex also does this truncation when reading the values back:

```cpp
    /// Access an adjacent edge id.  The eid value must be 0..3.
    EdgeId adjedge(int eid) const { return EdgeId((adjedges >> (2*eid)) & 3); }
```

It seems okay to consider this as working as expected for now. The C++ library does guard against some out-of-bad enum values when reading data back out so this should be okay for write_face.

Reply:
Ahh, if that is the case it might also be a good idea to read the data back and ensure the `EdgeId` values are in bounds too.  I'll go ahead and do that too, and we can always roll it back if that isn't wanted for whatever reason.